pydantic/config.py

```diff
@@ -34,185 +35,175 @@ class Extra(str, Enum):
-    json_loads: Callable[[str], Any] = json.loads  # TODO decide
-    json_dumps: Callable[..., str] = json.dumps  # TODO decide
-    json_encoders: Dict[Union[Type[Any], str, ForwardRef], Callable[..., Any]] = {}  # TODO decide
-    underscore_attrs_are_private: bool = False  # TODO remove
+    keep_untouched: Tuple[type, ...] = ()
+    json_loads: Callable[[str], Any] = json.loads
+    json_dumps: Callable[..., str] = json.dumps
+    json_encoders: Dict[Union[Type[Any], str, ForwardRef], Callable[..., Any]] = {}
     allow_inf_nan: bool = True
-
     strict: bool = False
+    copy_on_model_validation: Literal['none', 'deep', 'shallow'] = 'shallow'
+    post_init_call: Literal['before_validation', 'after_validation'] = 'before_validation'
 
-    # whether inherited models as fields should be reconstructed as base model,
-    # and whether such a copy should be shallow or deep
-    copy_on_model_validation: Literal['none', 'deep', 'shallow'] = 'shallow'  # TODO remove???
 
-    # whether `Union` should check all allowed types before even trying to coerce
-    smart_union: bool = False  # TODO remove
-    # whether dataclass `__post_init__` should be run before or after validation
-    post_init_call: Literal['before_validation', 'after_validation'] = 'before_validation'  # TODO remove
-
-    @classmethod
-    def get_field_info(cls, name: str) -> Dict[str, Any]:
-        """
-        Get properties of FieldInfo from the `fields` property of the config class.
-        """
-
-        fields_value = cls.fields.get(name)
-
-        if isinstance(fields_value, str):
-            field_info: Dict[str, Any] = {'alias': fields_value}
-        elif isinstance(fields_value, dict):
-            field_info = fields_value
-        else:
-            field_info = {}
-
-        if 'alias' in field_info:
-            field_info.setdefault('alias_priority', 2)
-
-        if field_info.get('alias_priority', 0) <= 1 and cls.alias_generator:
-            alias = cls.alias_generator(name)
-            if not isinstance(alias, str):
-                raise TypeError(f'Config.alias_generator must return str, not {alias.__class__}')
-            field_info.update(alias=alias, alias_priority=1)
-        return field_info
-
-    @classmethod
-    def prepare_field(cls, field: Any) -> None:
-        """
-        Optional hook to check or modify fields during model creation.
-        """
-        pass
-
-
-def get_config(config: Union[ConfigDict, Type[object], None]) -> Type[BaseConfig]:
+def get_config(config: Union[ConfigDict, Dict[str, Any], Type[Any], None]) -> ConfigDict:
     if config is None:
-        return BaseConfig
+        return ConfigDict()
 
-    else:
-        config_dict = (
-            config
-            if isinstance(config, dict)
-            else {k: getattr(config, k) for k in dir(config) if not k.startswith('__')}
+    if not isinstance(config, dict):
```

Comment:
I prefer to avoid unnecessary negatives, and also consolidate the `isinstance(config, dict)` checks here.

So something like 

```py
if isinstance(config, dict):
    config_dict = config
else:
    warnings.warn(...)
    config_dict = {k: getattr(config, k) for k in dir(config) if not k.startswith('__')}
```

?